josh-proxy/src/auth.rs

```diff
@@ -74,20 +74,34 @@ pub fn add_auth(token: &str) -> josh::JoshResult<Handle> {
 }
 
 pub async fn check_auth(url: &str, auth: &Handle, required: bool) -> josh::JoshResult<bool> {
+    use opentelemetry_semantic_conventions::trace::HTTP_RESPONSE_STATUS_CODE;
+
     if required && auth.hash.is_empty() {
         return Ok(false);
     }
 
-    if let Some(last) = AUTH_TIMERS.lock()?.get(&(url.to_string(), auth.clone())) {
+    let mut auth_timers = AUTH_TIMERS.lock().await;
```

Comment:
With the lock now spanning the whole function including the http request itself, I see a potential problem when multiple requests with different users arrive at the same time. Those will now be completely serial.
Might not be a big issue though.